pkg/prediction/dsp/prediction.go

```diff
@@ -145,28 +149,56 @@ func (p *periodicSignalPrediction) Run(stopCh <-chan struct{}) {
 		return
 	}
 
-	for {
-		// Waiting for a WithQuery request
-		queryExpr := p.qr.Read().(string)
-		klog.InfoS("Dsp received a WithQuery reques", "queryExpr", queryExpr)
-
-		go func(queryExpr string) {
-			ticker := time.NewTicker(p.modelConfig.UpdateInterval)
-			defer ticker.Stop()
-			for {
-				klog.V(4).InfoS("DSP: try to updateAggregateSignalsWithQuery", "queryExpr", queryExpr)
-				if err := p.updateAggregateSignalsWithQuery(queryExpr); err != nil {
-					klog.ErrorS(err, "Failed to updateAggregateSignalsWithQuery.")
+	go func() {
+		for {
+			// Waiting for a WithQuery request
+			queryExpr := <-p.withCh
+			if _, ok := p.stopChMap.Load(queryExpr); ok {
+				continue
+			}
+			klog.InfoS("Register a query expression for prediction.", "queryExpr", queryExpr)
```

Comment:
one common comment, add log level for debug info

Reply:
Added